src/java/main/org/apache/zookeeper/server/NettyServerCnxnFactory.java

```diff
@@ -524,17 +538,14 @@ public InetSocketAddress getLocalAddress() {
 
     private void addCnxn(NettyServerCnxn cnxn) {
         cnxns.add(cnxn);
-        synchronized (ipMap){
-            InetAddress addr =
-                ((InetSocketAddress)cnxn.channel.getRemoteAddress())
-                    .getAddress();
-            Set<NettyServerCnxn> s = ipMap.get(addr);
-            if (s == null) {
-                s = new HashSet<NettyServerCnxn>();
-            }
-            s.add(cnxn);
-            ipMap.put(addr,s);
+
+        InetAddress addr = ((InetSocketAddress)cnxn.channel.getRemoteAddress()).getAddress();
```

Comment:
Indeed, there's too much call for casting in this class already. Are you suggesting to extract this line into a method?